src/Shared/RequestDelegateFactoryUtilities.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Http
+{
+    // REVIEW: Better name?
+    internal static class RequestDelegateFactoryUtilities
```

Comment:
`TryParseMethodCache` or something like that? Naming a type utils makes it very easy for it be a dumping ground for unrelated stuff